File: DrivingRecorder-iOS/DrivingRecorder/Models/DrivingEvent.swift
import Foundation

/// 驾驶行为事件
struct DrivingEvent: Codable, Identifiable {
    let id: UUID
    let tripId: UUID
    let timestamp: Date
    let eventType: EventType
    let latitude: Double
    let longitude: Double
    let speedKmh: Double
    let heading: Double
    let severity: Double       // 0.0 - 1.0
    let description: String

    init(
        id: UUID = UUID(),
        tripId: UUID,
        timestamp: Date = Date(),
        eventType: EventType,
        latitude: Double,
        longitude: Double,
        speedKmh: Double,
        heading: Double,
        severity: Double,
        description: String
    ) {
        self.id = id
        self.tripId = tripId
        self.timestamp = timestamp
        self.eventType = eventType
        self.latitude = latitude
        self.longitude = longitude
        self.speedKmh = speedKmh
        self.heading = heading
        self.severity = severity
        self.description = description
    }
}

File: DrivingRecorder/app/src/main/java/com/drivingrecorder/domain/model/DrivingEvent.kt
package com.drivingrecorder.domain.model

/**
 * 驾驶行为事件
 */
data class DrivingEvent(
    val id: Long = 0,
    val tripId: Long = 0,
    val timestamp: Long,
    val eventType: EventType,
    val latitude: Double,
    val longitude: Double,
    val speed: Float,               // km/h
    val heading: Float,             // degrees
    val severity: Float,            // 0.0 - 1.0 严重程度
    val description: String
)
